libraries/AC_PID/AC_P_1D.cpp

```diff
@@ -0,0 +1,171 @@
+/// @file	AC_P_1D.cpp
+/// @brief	Generic P algorithm
+
+#include <AP_Math/AP_Math.h>
+#include "AC_P_1D.h"
+
+const AP_Param::GroupInfo AC_P_1D::var_info[] = {
+    // @Param: P
+    // @DisplayName: P Proportional Gain
+    // @Description: P Gain which produces an output value that is proportional to the current error value
+    AP_GROUPINFO("P",    0, AC_P_1D, _kp, 0),
+    AP_GROUPEND
+};
+
+// Constructor
+AC_P_1D::AC_P_1D(float initial_p, float dt) :
+    _dt(dt)
+{
+    // load parameter values from eeprom
+    AP_Param::setup_object_defaults(this, var_info);
+
+    _kp = initial_p;
+    _error = 0.0f;
+    _lim_err_neg = 0.0f;     // error value to enable filtering
+    _lim_err_pos = 0.0f;     // error value to enable filtering
+    _lim_out_neg = 0.0f;     // error value to enable filtering
+    _lim_out_pos = 0.0f;     // error value to enable filtering
+    _lim_D_Out = 10.0f;     // maximum first differential of output
+}
+
+// set_dt - set time step in seconds
+void AC_P_1D::set_dt(float dt)
+{
+    // set dt and calculate the input filter alpha
```

Comment:
Not calculating the input alpha.

Should be moved to the header IMO.

Reply:
Thanks, moved to header